### PR TITLE
React app for /users/{user_id} to the new function and Selenium

### PR DESCRIPTION
**Pull request by natni**

### CODE REVIEW

**Comment by natni** [comment]
Description
Closes #141 

This PR holds a script that fetches results for /users/userid API from the new Azure function and an investigation on a testing framework for Web UI which is **_Selenium_** 

**Testing Fetcing from Azure Function**

1. Navigate to ui/react-ui folder
2. Download packages and dependencies needed using `npm install`
3. Start the React app by typing `npm start`
4. When the app is successfully running on your localhost, please type _/reports/functionUserIdReport_ on your browser to view all the users fetched from the new function.
5. A table that shows results for user/{userid} should display like this 
Screen Shot 2021-02-16 at 5 14 06 PM

|**DATE**|**ACTIVITY**|**TIME**|
|---|:---|:---|
|02/07|Research on azure functions|2 hours|
|02/09|Created a table using 'react-bootstrap'|1 hour|







**Testing Selenium**

1. Download Chrome Driver from [here](https://chromedriver.storage.googleapis.com/index.html?path=88.0.4324.96/).
2. If you are on Mac copy and paste the **chromedriver** to this folder on your local machine /usr/local/bin
3. If you are on Windows copy and paste the **chromedriver** to this folder on your local machine C:\Windows
4. Navigate to ui/react-ui folder and download packages and dependencies needed using `npm install`
5. Navigate to ui/react-ui/components/seleniumExample folder and run `node testUi.js`
6. You'll be able to see a basic test that checks if artillery results are there or not. It opens a chrome browser that is controlled by an automated test software.
Screen Shot 2021-02-18 at 7 15 58 PM



|**DATE**|**ACTIVITY**|**TIME**|
|---|:---|:---|
|02/08| Research on Testing Framework |3 hours|
|02/09| Implemented a simple script for testing |1 hour|

File: ui/react-ui/src/components/routes/reports/functionReport/FunctionUserIdReport.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React, { useState, useEffect } from 'react';
+import { JsonToTable } from "react-json-to-table";
+
+function FunctionUserIdReport() {
+    const [jsonReport, setJsonReport] = useState();
+    const [showJson, setshowJson] = useState(false);
+    const jsonUrl = "https://nsc-fun-dev-usw2-thursday.azurewebsites.net/api/users";
```
Comment (farhadbahrehmandhenry) [inline ui/react-ui/src/components/routes/reports/functionReport/FunctionUserIdReport.js]: seems you are implementing /users instead of /users/{user_id}

**Review by farhadbahrehmandhenry — COMMENTED** [review]
we already have a component that fetches data for /users/{user_id} and renders it on the page. (the component name is "userid") I'm wondering why we need another component to do the same thing here.

File: ui/react-ui/src/components/routes/reports/functionReport/FunctionUserIdReport.js (hunk 1 above)
Comment (natni) [reply ui/react-ui/src/components/routes/reports/functionReport/FunctionUserIdReport.js]: Solved it now...thanks:-)

**Review by farhadbahrehmandhenry — APPROVED** [review]

**Pull request closed without merging**